### PR TITLE
Update integration to new installation methods.

### PR DESCRIPTION
**Pull request by pszczesniak**
This PR should update integration to new installation methods of `CKEditor5`.

### CODE REVIEW

**Comment by Witoso** [comment]
What changes:   
\* the current installation methods (getting things from src and DLLs) are still supported but will be put on the sunset path (for around a year)  
\* we added new installation method, which basically bundles the js code to `dist`.

File: package.json [hunk 1]
```diff
@@ -9,38 +9,46 @@
...
-    "@ckeditor/ckeditor5-editor-classic": ">=16.0.0",
-    "@ckeditor/ckeditor5-essentials": ">=16.0.0",
-    "@ckeditor/ckeditor5-heading": ">=16.0.0",
-    "@ckeditor/ckeditor5-list": ">=16.0.0",
-    "@ckeditor/ckeditor5-paragraph": ">=16.0.0",
-    "@ckeditor/ckeditor5-theme-lark": ">=16.0.0",
+    "@ckeditor/ckeditor5-dev-utils": ">=40.2.0",
+    "@ckeditor/ckeditor5-dev-build-tools": "^40.2.0",
+    "@ckeditor/ckeditor5-package-tools": "^1.1.0",
+    "ckeditor5": "nightly",
```
Comment (pszczesniak) [inline package.json:51]: After the release the `CKEditor5` in version 42 it should be changed from nightly to the latest version (`>=42.0.0`).

File: package.json [hunk 2]
```diff
@@ -51,7 +59,7 @@
     "webpack": "^5.82.0",
     "webpack-cli": "^5.1.1"
   },
-  "dependencies": {
-    "ckeditor5": ">=16.0.0"
+  "peerDependencies": {
+    "ckeditor5": ">=42.0.0 || ^0.0.0-nightly"
```
Comment (Bohdan-Chui) [inline package.json:63]: Should we remove `|| ^0.0.0-nightly` after the CKEditor5 in version 42?

File: package.json [hunk 3]
```diff
@@ -9,38 +9,46 @@
   "scripts": {
     "start": "./node_modules/.bin/webpack --mode development -w",
     "build": "./node_modules/.bin/webpack --mode production",
-    "test": "concurrently --kill-others --success \"first\" \"node --max_old_space_size=4096 node_modules/@ckeditor/ckeditor5-dev-tests/bin/testautomated.js -c --files=ckeditor5\" \"http-server -c-0 --cors -p 3000 -s\""
+    "build:dist": "node ./scripts/build-dist.mjs",
+    "prepare": "npm run build && npm run build:dist",
```
Comment (Bohdan-Chui) [inline package.json:13]: Did we really understand correctly that you need to use our `npm run build` before your `npm run build:dist`?

**Comment by pszczesniak** [comment]
Imports from path like `'ckeditor5/src/[NAME].js'` is to support DLL builds.

Our new tool (which is used in `scripts/build-dist.mjs`) is changing during the build process all the imports paths to support new installations methods.

File: package.json (hunk 3 above)
Comment (pszczesniak) [reply package.json:13]: As far as i can see, the `build` won't be published, so the answer is no. Will remove it.

Comment (pszczesniak) [reply package.json]: Addressed in pszczesniak@cb18236.

File: package.json (hunk 2 above)
Comment (pszczesniak) [reply package.json:63]: Yes.

Comment (pszczesniak) [reply package.json:63]: @Bohdan-Chui After a consideration: it can stay as it is now. It will allows to run this plugin with nightly builds.

**Review by Bohdan-Chui — APPROVED** [review]
Looks good!

File: README.md [hunk 4]
```diff
@@ -70,6 +70,23 @@ WProofreader plugin for CKEditor 5 inherits all functionality of the WProofreade
 
 	Unlike the Cloud-based version, the `serviceId` parameter is not used here. Instead, it is required to specify the path to the backend entry point hosted on the client’s infrastructure.
 
+	For WProofreader that uses `CKEditor5` without legacy methods of installation:
```
Comment (Reinmar) [inline README.md:73]: "CKEditor 5"

No need to wrap it with backticks.

File: package.json [hunk 5]
```diff
@@ -9,38 +9,46 @@
...
-    "@ckeditor/ckeditor5-basic-styles": ">=16.0.0",
-    "@ckeditor/ckeditor5-dev-tests": ">=16.0.0",
-    "@ckeditor/ckeditor5-dev-utils": ">=13.0.0",
-    "@ckeditor/ckeditor5-editor-classic": ">=16.0.0",
-    "@ckeditor/ckeditor5-essentials": ">=16.0.0",
-    "@ckeditor/ckeditor5-heading": ">=16.0.0",
-    "@ckeditor/ckeditor5-list": ">=16.0.0",
-    "@ckeditor/ckeditor5-paragraph": ">=16.0.0",
-    "@ckeditor/ckeditor5-theme-lark": ">=16.0.0",
+    "@ckeditor/ckeditor5-dev-utils": ">=40.2.0",
```
Comment (Reinmar) [inline package.json:48]: Shouldn't we propose a consistency on how we install dev-utils and dev-build-tools? Now, one rely on `>=` and the other on `^`.

File: samples/app.js [hunk 6]
```diff
@@ -1,12 +1,12 @@
-import ClassicEditorBase from '@ckeditor/ckeditor5-editor-classic/src/classiceditor';
-import Essentials from '@ckeditor/ckeditor5-essentials/src/essentials';
-import Paragraph from '@ckeditor/ckeditor5-paragraph/src/paragraph';
-import Heading from '@ckeditor/ckeditor5-heading/src/heading';
-import List from '@ckeditor/ckeditor5-list/src/list';
-import Bold from '@ckeditor/ckeditor5-basic-styles/src/bold';
-import Italic from '@ckeditor/ckeditor5-basic-styles/src/italic';
-import WProofreader from '../src/wproofreader';
+import { ClassicEditor as ClassicEditorBase } from '@ckeditor/ckeditor5-editor-classic';
+import { Essentials } from '@ckeditor/ckeditor5-essentials';
```
Comment (Reinmar) [inline samples/app.js:2]: Shouldn't we recommend imports from `ckeditor5`?

File: samples/multi-root-editor/app.js [hunk 7]
```diff
@@ -1,12 +1,13 @@
-import MultirootEditorBase from './multi-root-editor';
-import Essentials from '@ckeditor/ckeditor5-essentials/src/essentials';
-import Paragraph from '@ckeditor/ckeditor5-paragraph/src/paragraph';
-import Heading from '@ckeditor/ckeditor5-heading/src/heading';
-import List from '@ckeditor/ckeditor5-list/src/list';
-import Bold from '@ckeditor/ckeditor5-basic-styles/src/bold';
-import Italic from '@ckeditor/ckeditor5-basic-styles/src/italic';
-import WProofreader from '../../src/wproofreader';
+import MultirootEditorBase from './multi-root-editor.js';
+import { Essentials } from '@ckeditor/ckeditor5-essentials';
```
Comment (Reinmar) [inline samples/multi-root-editor/app.js:2]: Shouldn't we recommend imports from `ckeditor5`?

File: src/wproofreader.js [hunk 8]
```diff
@@ -116,7 +117,7 @@ export default class WProofreader extends Plugin {
 	 * @private
 	 */
 	_setAutoStartup() {
-		if (!this._userOptions.hasOwnProperty('autoStartup')) {
+		if (!Object.prototype.hasOwnProperty.call(this._userOptions, 'autoStartup')) {
```
Comment (Reinmar) [inline src/wproofreader.js:120]: Why are we changing this?

File: tests/mocks/mock-collaboration-editing.js [hunk 9]
```diff
@@ -1,4 +1,6 @@
-import Plugin from '@ckeditor/ckeditor5-core/src/plugin';
+
+// import Plugin from '@ckeditor/ckeditor5-core/src/plugin.js';
```
Comment (Reinmar) [inline tests/mocks/mock-collaboration-editing.js:2]: remove

Comment (pszczesniak) [reply tests/mocks/mock-collaboration-editing.js]: Addressed in pszczesniak@380dfb3.

File: README.md (hunk 4 above)
Comment (pszczesniak) [reply README.md]: Addressed in pszczesniak@b9038d3.

File: package.json (hunk 5 above)
Comment (pszczesniak) [reply package.json]: Addressed in pszczesniak@614eb06.

File: README.md [hunk 10]
```diff
@@ -70,6 +70,23 @@ WProofreader plugin for CKEditor 5 inherits all functionality of the WProofreade
...
+	For WProofreader that uses "CKEditor 5" without legacy methods of installation:
+
+	```js
+	import { ClassicEditor } from 'ckeditor5';
+	import { WProofreader } from '@webspellchecker/wproofreader-ckeditor5';
+	...
+
+	ClassicEditor
+		.create( editorElement, {
+			plugins: [ ..., WProofreader],
```
Comment (Reinmar) [inline README.md:82]: space before `]` missing

File: README.md [hunk 11]
```diff
@@ -70,6 +70,23 @@ WProofreader plugin for CKEditor 5 inherits all functionality of the WProofreade
...
+			toolbar: [ ..., 'wproofreader'],
```
Comment (Reinmar) [inline README.md:83]: space before `]` missing

File: README.md (hunk 10 above)
Comment (pszczesniak) [reply README.md]: Addressed in pszczesniak@f609fc0.

File: README.md (hunk 11 above)
Comment (pszczesniak) [reply README.md]: Addressed in pszczesniak@f609fc0.

File: samples/app.js (hunk 6 above)
Comment (pszczesniak) [reply samples/app.js]: Addressed in https://github.com/pszczesniak/wproofreader-ckeditor5/commit/6ec993e7a998c6ba759c5732fa35896755904fe0.

File: samples/multi-root-editor/app.js (hunk 7 above)
Comment (pszczesniak) [reply samples/multi-root-editor/app.js]: Addressed in https://github.com/pszczesniak/wproofreader-ckeditor5/commit/6ec993e7a998c6ba759c5732fa35896755904fe0.

File: tests/wproofreader.js [hunk 12]
```diff
@@ -28,6 +28,11 @@ describe('WProofreader', () => {
 		return element.remove();
 	});
 
+		it('should be loaded', () => {
```
Comment (Bohdan-Chui) [inline tests/wproofreader.js:31]: Do we really need this test case? If yes, please fix tabulation.

Comment (pszczesniak) [reply tests/wproofreader.js:31]: A leftover, good catch, thank You!

Comment (pszczesniak) [reply tests/wproofreader.js]: Addressed in pszczesniak@78bd87a.

**Pull request merged**